shell/platform/darwin/ios/framework/Source/UIViewController+FlutterScreenAndSceneIfLoaded.mm

```diff
@@ -0,0 +1,30 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/ios/framework/Source/UIViewController+FlutterScreenAndSceneIfLoaded.h"
+
+#include "flutter/fml/logging.h"
+
+@implementation UIViewController (FlutterScreenAndSceneIfLoaded)
+
+- (UIWindowScene*)flutterWindowSceneIfViewLoaded {
+  if (self.viewIfLoaded == nil) {
+    FML_LOG(WARNING) << "Trying to access the window scene before the view is loaded.";
+    return nil;
+  }
+  return self.viewIfLoaded.window.windowScene;
+}
+
+- (UIScreen*)flutterScreenIfViewLoaded {
```

Comment:
Same question about naming convention.

Reply:
I guess I'm not clear the difference between class method and global function in this case.

Reply:
A global function would just work, but there are a few benefits of using categories in this particular case. Just from top of my head - 

1. it's more idiomatic objc code, and more consistent with Apple's convention (e.g. `vc.viewIfLoaded`, rather than `viewIfLoadedForVC(vc)`)

2. It can be stubbed for tests. Category methods use "message dispatch", meaning that the implementation can be swizzled out during run-time (basically how `OCMock` works internally)